sycl/include/sycl/exception.hpp

```diff
@@ -136,9 +136,8 @@ class __SYCL_EXPORT exception : public virtual std::exception {
             const char *WhatArg);
 };
 
-class __SYCL2020_DEPRECATED(
-    "use sycl::exception with sycl::errc::runtime instead.") runtime_error
-    : public exception {
+class __SYCL_DEPRECATED("use sycl::exception with sycl::errc::runtime instead.")
```

Comment:
Strictly speaking, this is an incorrect change. Those classes are still valid in SYCL 1.2.1 mode which can still be selected with our implementation:

https://github.com/intel/llvm/blob/c821dc934dc7934b0209b5d3f88a280bbaa7145c/clang/include/clang/Driver/Options.td#L8316-L8321

I'm sure that it has been completely broken for a while now, but I would prefer to have a separate PR that completely removes it consistently from everywhere